ext/sockets/sockets.c

```diff
@@ -1845,12 +1847,12 @@ PHP_FUNCTION(socket_set_option)
 			opt_ht = Z_ARRVAL_P(arg4);
 
 			if ((l_onoff = zend_hash_str_find(opt_ht, l_onoff_key, sizeof(l_onoff_key) - 1)) == NULL) {
-				php_error_docref(NULL, E_WARNING, "No key \"%s\" passed in optval", l_onoff_key);
-				RETURN_FALSE;
+				zend_argument_value_error(4, "must have key \"%s\"", l_onoff_key);
```

Comment:
would be `must have the "%s" key` any better?

Reply:
Hum, I don't know, I'll leave it as it for the moment

Reply:
Fine!